### PR TITLE
fix: boojum integration nightly rust

### PR DESCRIPTION
**Pull request by AntonD3**
# What :computer: 
* use nightly rust for the github workflows

# Why :hand:
* the `boojum-integration` branch requires nightly rust

# Evidence :camera:
The successful run of the release workflow: https://github.com/AntonD3/era-test-node/actions/runs/6650532245

### CODE REVIEW

**Comment by MexicanAce** [comment]
We should avoid adding this to your branch. If you need a binary for `boojum-integration`, we should pull it down and generate it in the CI of the downstream repos (e.g. `era-contracts` and `era-system-contracts`).

**Comment by AntonD3** [comment]
> We should avoid adding this to your branch. If you need a binary for `boojum-integration`, we should pull it down and generate it in the CI of the downstream repos (e.g. `era-contracts` and `era-system-contracts`).

Why we should avoid it? That's not only related to the system contracts. `boojum-integration` is a long-living branch, and at some point, we should merge it into the main, give it to the users, and nightly is required by the `zksync-era` boojum version.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -44,23 +44,23 @@ jobs:
...
+        run: cargo +nightly clean
 
       # ==============================
       #       Builds
       # ==============================
 
       - name: Install cross from source
         run: |
-          cargo install --git https://github.com/cross-rs/cross cross --branch main
+          cargo +nightly install --git https://github.com/cross-rs/cross cross --branch main
```
Comment (MexicanAce) [inline .github/workflows/release.yml:63]: Please add Verification Evidence to the Pull Request description showing that releases work end-to-end with Nightly builds. We typically do this by forking and running a release in a forked branch

**Comment by MexicanAce** [comment]
> nightly is required by the `zksync-era` boojum version

Oof, that's rough. I'm guessing the plan is to merge `boojum-integration` into `main` when the `zksync-era` boojum version is merged also... and that's going to be merged with Nightly enabled.

Ok, I think I see the broader picture. Sounds like this branch should support Boojum in the same way that `zksync-era` supports Boojum. And for now, that means nightly builds. 🤯 😅

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -47,7 +46,7 @@ jobs:
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
         with:
-          toolchain: stable
+          toolchain: nightly
 
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2
```
Comment (MexicanAce) [inline .github/workflows/checks.yaml:52]: Not to self, this Rust Cache might be getting "cache busted" when a branch uses the nightly packages and the `main` branch uses stable

**Comment by AntonD3** [comment]
> Ok, I think I see the broader picture. Sounds like this branch should support Boojum in the same way that `zksync-era` supports Boojum. And for now, that means nightly builds. 🤯 😅

Yes, I am not sure it will removed, but if so, we can always restore stable rust here as well

File: .github/workflows/release.yml (hunk 1 above)
Comment (AntonD3) [reply .github/workflows/release.yml:63]: https://github.com/AntonD3/era-test-node/actions/runs/6507401464

**Comment by AntonD3** [comment]
@MexicanAce Do you know why unit-tests CI failed? It looks like it was canceled, do we have any limits?

**Comment by MexicanAce** [comment]
>Do you know why unit-tests CI failed? It looks like it was canceled, do we have any limits?

VERY strange, `143` definitely makes it look like it was killed/stopped, but I see no evidence of anyone/anything initiating the termination. We don't have limits, in fact some of our release builds take 43+ minutes, so a 15 minute build should be just fine.

I'm taking a note, and kicking them off again.

**Comment by AntonD3** [comment]
@MexicanAce I think the issue can be caused by the runner limits. Do you know what the machine parameters are?

**Comment by MexicanAce** [comment]
@AntonD3 I'm unsure what the exact limits are as we're using `ubuntu-latest`, which is a VM owned/operated by GitHub

**Comment by AntonD3** [comment]
Well, if [these limits](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) are true for us, the cause of the problem can be the SSD space limit.

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -21,15 +21,15 @@ jobs:
       uses: actions/checkout@v3
 
     - name: Install Rust
-      uses: dtolnay/rust-toolchain@stable
-      with: 
-        toolchain: stable
+      uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        toolchain: nightly-2023-04-17
```
Comment (dutterbutter) [inline .github/workflows/tests.yaml:26]: If we are switching to this rust toolchain action can you update the other workflows as well. In addition, this action has `Swatinem/rust-cache` built in, can you update so we don't have redundant cache.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -44,8 +44,8 @@ jobs:
     steps:
       - name: Checkout sources
         uses: actions/checkout@v3
-      - name: Get latest version of stable Rust
-        run: rustup update stable
+      - name: Set version of nightly Rust
+        run: rustup default nightly-2023-04-17
```
Comment (AntonD3) [inline .github/workflows/release.yml:48]: @dutterbutter I can not find the place where the `rustup` is installed for this workflow. Maybe you know, is it preinstalled?

**Comment by AntonD3** [comment]
@MexicanAce with this rust action it works stable as you can see. Is [such result](https://github.com/AntonD3/era-test-node/actions/runs/6606361445) for the release workflow fine?

**Comment by MexicanAce** [comment]
>Is [such result](https://github.com/AntonD3/era-test-node/actions/runs/6606361445) for the release workflow fine?
@AntonD3 please merge latest from upstream `main` to resolve this issue of releases

**Comment by AntonD3** [comment]
https://github.com/AntonD3/era-test-node/actions/runs/6650532245

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**